source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRImageSetUICollectionView.mm

```diff
@@ -105,26 +104,37 @@ - (CGSize)intrinsicContentSize
 
 - (CGSize)collectionViewContentSize
 {
-    size_t cellCounts = _imgSet->GetImages().size();
+    NSUInteger cellCounts = _imgSet->GetImages().size();
     CGSize imageSize = ((UICollectionViewFlowLayout *)self.collectionViewLayout).itemSize;
-    float spacing = ((UICollectionViewFlowLayout *)self.collectionViewLayout).minimumInteritemSpacing;
-    float lineSpacing = ((UICollectionViewFlowLayout *)self.collectionViewLayout).minimumLineSpacing;
+    CGFloat spacing = ((UICollectionViewFlowLayout *)self.collectionViewLayout).minimumInteritemSpacing;
+    CGFloat lineSpacing = ((UICollectionViewFlowLayout *)self.collectionViewLayout).minimumLineSpacing;
 
     // sanity check
-    if (!imageSize.width || !self.frame.size.width || !cellCounts) {
+    if (!imageSize.width || !cellCounts) {
         return CGSizeMake(0, 0);
     }
-    float frameWidth = self.frame.size.width;
-    float imageWidthWithSpacing = imageSize.width + spacing;
+    CGFloat frameWidth = self.frame.size.width;
+    CGFloat imageWidthWithSpacing = imageSize.width + spacing;
 
     // if there is spacing to the right edge, it's o.k.
-    int numbersOfItemsInRow = frameWidth / imageWidthWithSpacing;
+    NSUInteger numbersOfItemsInRow = (frameWidth) ? (frameWidth) / imageWidthWithSpacing : cellCounts;
     // if addtional image can be fit by removing spacing, do so
     if (numbersOfItemsInRow * imageWidthWithSpacing + imageSize.width <= frameWidth) {
         numbersOfItemsInRow++;
     }
 
-    int numbersOfRows = ceil(((float)cellCounts) / numbersOfItemsInRow);
-    return CGSizeMake(self.frame.size.width, (numbersOfRows) * (imageSize.height) + (numbersOfRows - 1) * lineSpacing);
+    NSUInteger numbersOfRows = ceil(((CGFloat)cellCounts) / numbersOfItemsInRow);
```

Comment:
is there a risk that `numbersOfItemsInRow` could be `0`?

Reply:
thanks made the `numberOfItemsInRow` to be at least 1.